lte/gateway/python/magma/subscriberdb/store/sqlite.py

```diff
@@ -90,10 +96,19 @@ def edit_subscriber(self, subscriber_id):
             subscriber_data.ParseFromString(row[0])
             yield subscriber_data
             data_str = subscriber_data.SerializeToString()
-            self.conn.execute(
-                "UPDATE subscriberdb SET data = ? " "WHERE subscriber_id = ?",
-                (data_str, subscriber_id),
-            )
+            try:
+                self.conn.execute(
+                    "UPDATE subscriberdb SET data = ? " "WHERE subscriber_id = ?",
+                    (data_str, subscriber_id),
+                )
+            except sqlite3.OperationalError:
+                logging.error("sqlite db error for subscriber: %s", subscriber_id)
+                time.sleep(random.randint(RAND_LOWERBOUND, RAND_UPPERBOUND)/100)
+                # Try one more time after random wait
+                self.conn.execute(
```

Comment:
What happens if you hit exception here ? Can we perhaps wrap this in a async block ? Does this failure trigger a failure at the caller ?

Reply:
It is caught and prints a trace in syslog.